--- origsrc/silc-toolkit-1.1.9/configure.ac	2009-01-08 08:29:31.000000000 -0600
+++ src/silc-toolkit-1.1.9/configure.ac	2009-05-03 17:23:13.484250000 -0500
@@ -360,23 +360,27 @@ have_assembler=false
 if test x$want_asm = xtrue; then
   AC_PATH_PROG([NASM], [nasm], [no])
   if test "x$NASM" != "xno"; then
-    if test x$cpu_x86_64 = xtrue; then
-      SILC_ASSEMBLER="$NASM -O2 -felf64"
-    else
-      SILC_ASSEMBLER="$NASM -O2 -felf"
-    fi
+    SILC_ASSEMBLER="$NASM -O2"
     have_assembler=true
   fi
 
   AC_PATH_PROG([YASM], [yasm], [no])
   if test "x$YASM" != "xno"; then
-    if test x$cpu_x86_64 = xtrue; then
-      SILC_ASSEMBLER="$YASM -Xgnu -felf64"
-    else
-      SILC_ASSEMBLER="$YASM -Xgnu -felf"
-    fi
+    SILC_ASSEMBLER="$YASM -Xgnu"
     have_assembler=true
   fi
+
+  case "$host" in
+    *-*-cygwin*|*-*-mingw*)
+      SILC_ASSEMBLER="$SILC_ASSEMBLER -fwin32"
+      ;;
+    x86_64-*)
+      SILC_ASSEMBLER="$SILC_ASSEMBLER -felf64"
+      ;;
+    *)
+      SILC_ASSEMBLER="$SILC_ASSEMBLER -felf"
+      ;;
+  esac
 fi
 AC_SUBST(SILC_ASSEMBLER)
 
